File: app/src/main/java/com/kindleassistant/di/AppComponent.java
package com.kindleassistant.di;


import com.kindleassistant.App;
import com.kindleassistant.sender.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, PresenterModule.class})
public interface AppComponent {
    void inject(App app);

    void inject(MainActivity activity);
}

File: app/src/main/java/com/kindleassistant/di/AppComponent.kt
package com.kindleassistant.di


import com.kindleassistant.App
import com.kindleassistant.sender.MainActivity
import com.kindleassistant.upload.UploadActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [(AppModule::class), (PresenterModule::class), (NetworkModule::class)])
interface AppComponent {
    fun inject(app: App)

    fun inject(activity: MainActivity)
    fun inject(activity: UploadActivity)
}